src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -274,6 +275,21 @@ public static T[] AsArray<T>(this IEnumerable<T> source)
             return builder.ToImmutableAndFree();
         }
 
+        /// <summary>
+        /// Maps an immutable array through a function that returns ValueTask, returning the new ImmutableArray.
+        /// </summary>
+        public static async ValueTask<ImmutableArray<TResult>> SelectAsArrayAsync<TItem, TResult>(this IEnumerable<TItem> source, Func<TItem, ValueTask<TResult>> selector)
+        {
+            var builder = ArrayBuilder<TResult>.GetInstance();
+
+            foreach (var item in source)
+            {
+                builder.Add(await selector(item).ConfigureAwait(false));
+            }
+
+            return builder.ToImmutableAndFree();
+        }
```

Comment:
do we not already have this?

Reply:
Not a variant on Enumerable, but on ImmutableArray.